### PR TITLE
[FEAT]: huggingface integration

### PR DESCRIPTION
**Pull request by universalmind303**
added a few public sample files to my personal huggingface account

```py
df = daft.read_csv("hf://datasets/universalmind303/daft-docs/iris.csv")
```

### CODE REVIEW

File: src/common/io-config/src/python.rs [hunk 1]
```diff
@@ -136,7 +136,7 @@ pub struct IOConfig {
 ///     user_agent (str, optional): The value for the user-agent header, defaults to "daft/{__version__}" if not provided
```
Comment (kevinzwang) [inline src/common/io-config/src/python.rs:136]: Please add docs for `bearer_token` here

File: src/daft-io/src/lib.rs [hunk 2]
```diff
@@ -367,6 +384,9 @@ pub fn parse_url(input: &str) -> Result<(SourceType, Cow<'_, str>)> {
                 msg: "Could not convert expanded path to string".to_string(),
             });
     }
+    if input.starts_with("hf://") {
+        return Ok((SourceType::HF, Cow::Borrowed(input)));
+    }
```
Comment (kevinzwang) [inline src/daft-io/src/lib.rs:389]: Is there a reason for having this as well as the match on `scheme` on like 409?

**Review by kevinzwang — COMMENTED** [review]
Nice work! This will definitely be useful for ML workflows.

One more note before merging: could you add a page to our integrations docs talking about how to use this feature? Some quick examples on using files from huggingface as well as how to authenticate using their user access tokens by passing it in as a bearer token.

File: src/daft-io/src/lib.rs (hunk 2 above)
Comment (universalmind303) [reply src/daft-io/src/lib.rs:389]: just tested without this, and it  seems to work fine, will remove!

**Review by kevinzwang — APPROVED** [review]
LGTM!

**Pull request merged**